src/stream/src/executor/chain.rs

```diff
@@ -78,29 +71,11 @@ impl ChainExecutor {
 
         // 1. Poll the upstream to get the first barrier.
         let barrier = expect_first_barrier(&mut upstream).await?;
-        let prev_epoch = barrier.epoch.prev;
-
-        // If the barrier is a conf change of creating this mview, init snapshot from its epoch
-        // and begin to consume the snapshot.
-        // Otherwise, it means we've recovered and the snapshot is already consumed.
-        let to_consume_snapshot = barrier.is_add_dispatcher(self.actor_id);
 
         // The first barrier message should be propagated.
         yield Message::Barrier(barrier);
 
-        // 2. Consume the snapshot if needed. Note that the snapshot is already projected, so
-        // there's no mapping required.
-        if to_consume_snapshot {
-            // Init the snapshot with reading epoch.
-            let snapshot = self.snapshot.execute_with_epoch(prev_epoch);
-
-            #[for_await]
-            for msg in snapshot {
-                yield msg?;
-            }
-        }
-
-        // 3. Continuously consume the upstream. Report that we've finished the creation on the
+        // 2. Continuously consume the upstream. Report that we've finished the creation on the
```

Comment:
Actually we need something that can pass the stream message directly from the upstream index to the downstream, but don't need to consume the snapshot. That's one of the problems I want to discuss with you.

Reply:
"Chain" is originally designed to chain historical snapshot with change stream after that. 

As I proposed before, recommend abandoning the obscure name "Chain" and using some more specific name like "StreamTableScan". cc. @BugenZhao @TennyZhuang

Reply:
> recommend abandoning the obscure name "Chain" and using some more specific name like "StreamTableScan" or some better names

weakly +1, because we can not support any batch query in the one unique physical "Chain" executor. I think the Chain is more like a logical operator.